TrackletAlgorithm/MatchEngine.h

```diff
@@ -104,18 +67,225 @@ constexpr unsigned int kZAdjustment = 8;
 
 /////////////////////////////
 // -- MATCH ENGINE FUNCTIONS
-void readTable(bool table[LSIZE]);
+inline void readTable(ap_uint<1> table[LSIZE]) {
+	ap_uint<1> tmp[LSIZE]=
+	#if LAYER == 1
+#include "../emData/ME/tables/METable_L1.tab"
+	#elif LAYER == 2
+#include "../emData/ME/tables/METable_L2.tab"
+	#elif LAYER == 3
+#include "../emData/ME/tables/METable_L3.tab"
+	#elif LAYER == 4
+#include "../emData/ME/tables/METable_L4.tab"
+	#elif LAYER == 5
+#include "../emData/ME/tables/METable_L5.tab"
+	#elif LAYER == 6
+#include "../emData/ME/tables/METable_L6.tab"
+	#else
+{};
+	#endif
+
+	for (int i=0;i<LSIZE;i++){
+		table[i]=tmp[i];
+	}
+}
 
 //template<int L, regionType VMSMEType>
-template<int L, int VMSMEType>
+template<int L, int VMSMEType, int VMPMEType>
```

Comment:
@aperloff I guess we need `VMPMEType` to differentiate between the barrel and the disks, but do we actually need `VMSMEType`? It looks like it can be determined from the value of `L`.

Reply:
It's needed for the template of one of the parameters in the `MachEngine` function (https://github.com/cms-L1TK/firmware-hls/pull/183/files#diff-8c6bdc6c6e457d4f237d95aeb3341adffd3be2757f53f1744c72ea6852ce4af8R96). Unless this is pre-computed in Constants.h it needs to be a template parameter. Would it be okay with you if we defined the following at https://github.com/cms-L1TK/firmware-hls/blob/master/TrackletAlgorithm/Constants.h#L132?
```c++
// List of layer and disk types
constexp unsigned int layerDiskType[N_LAYER + N_DISK] = {BARRELPS, BARRELPS, BARRELPS, BARREL2S, BARREL2S, BARREL2S, DISK, DISK, DISK, DISK, DISK};
```

Reply:
Yeah I think that's OK. But I also think keeping `VMPMEType` as a parameter is OK. I just didn't notice that it's used in the function arguments.

Reply:
Done